js/src/offcanvas.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * --------------------------------------------------------------------------
+ * Bootstrap (v4.3.1): offcanvas.js
+ * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
+ * --------------------------------------------------------------------------
+ */
+
+import { getSelectorFromElement, jQuery as $ } from './util/index'
+
+import Data from './dom/data'
+import EventHandler from './dom/event-handler'
+import SelectorEngine from './dom/selector-engine'
+
+/**
+ * ------------------------------------------------------------------------
+ * Constants
+ * ------------------------------------------------------------------------
+ */
+
+const NAME = 'offcanvas'
+const VERSION = '4.3.1'
+const DATA_KEY = 'bs.offcanvas'
+const EVENT_KEY = `.${DATA_KEY}`
+const DATA_API_KEY = '.data-api'
+
+const Selector = {
+  DATA_DISMISS: '[data-dismiss="offcanvas"]',
+  DATA_TOGGLE: '[data-toggle="offcanvas"]'
+}
+
+const Event = {
+  SHOW: `show${EVENT_KEY}`,
+  SHOWN: `shown${EVENT_KEY}`,
+  HIDE: `hide${EVENT_KEY}`,
+  HIDDEN: `hidden${EVENT_KEY}`,
+  CLICK_DATA_API: `click${EVENT_KEY}${DATA_API_KEY}`,
+  CLICK_DISMISS: `click.dismiss${EVENT_KEY}`
+}
+
+const ClassName = {
+  SHOW: 'show'
+}
+
+/**
+ * ------------------------------------------------------------------------
+ * Class Definition
+ * ------------------------------------------------------------------------
+ */
+
+class Offcanvas {
+  constructor(element) {
+    this._element = element
+    this._isShown = element.classList.contains(ClassName.SHOW)
+    if (this._element) {
+      Data.setData(element, DATA_KEY, this)
+    }
+
+    EventHandler.on(this._element, Event.CLICK_DISMISS, Selector.DATA_DISMISS, event => this.hide(event))
+    EventHandler.on(document, Event.CLICK_DATA_API, event => {
+      const target = Offcanvas._getParentFromElement(event.target)
```

Comment:
I'm more in favor of adding a css class to the body element to detect when a OffCanvas is open, it'll be faster to check if the body contains this class, and if it's the case we can try to figure out the parent of the target.

Reply:
Please be comfortable, to make corrections, as you have better knowledge of the code functionalities